### PR TITLE
Refactoring Home Component

### PR DESCRIPTION
**Pull request by DBULL7**
## What's this PR do?
Broke my Home Component into two components. Created a helper file that each component will use since they make the same search call. The helper manages a 150 line chain of scrubbers and API calls some of which will need refactoring. The helper can then be setup to dispatch to the Redux store. 

[Helper File to Handle API Call](https://github.com/DBULL7/Unavee/issues/104)
[Split Home Component Into Two Componets](https://github.com/DBULL7/Unavee/issues/101)

## Pre-Merge TODOs
 - [x] Add follow up issues?
   [Make Dispatch from Helper File](https://github.com/DBULL7/Unavee/issues/105)
  [Dispatch the search results to the Redux store](https://github.com/DBULL7/Unavee/issues/100)
 - [x] 2 PR reviews completed

## Where should the reviewer start?
The Home.js file. I essentially just broke the code from it out into another component and helper file. 

_What should the reviewer look at. Are there any potential red flags or concerns you have? How will the reviewer see the code (can they run it locally, etc?)_

The scrubSearch function in the helper file needs refactoring. 

## Any background context you want to provide?
The Home.js file was over 500 lines this morning. It is now down to 340 lines.

### CODE REVIEW

**Review by cbandrow-zz — COMMENTED** [review]
Some good looking refactors there dude. looks much cleaner.

**Review by jennPeavler — COMMENTED** [review]
Looking at the HomeHelper:

Get rid of all the console logs and the const named test, test2 or anything to do with test.  Then we can talk ;)

There's still a lot of refactoring to be done.  It seems like there is pretty complex functionality within the HomeHelper so if you can dry up your code a bit, it would help me a lot to follow along.  That way I can better make suggestions on how to refactor/reorganize.

File: app/Components/Helpers/HomeHelper.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
+ export const checkDatabaseForSearch = (search, setState, scrubbedTweets) => {
+  fetch(`/api/v1/${search}`)
+  .then(res => res.json())
+  .then(data => {
+    console.log(data)
```
Comment (rrgayhart) [inline app/Components/Helpers/HomeHelper.js:5]: Like Jenn said. Ain't nobody got time for console.logs

File: app/Components/Helpers/HomeHelper.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  if(demographics) {
+    setState({location: demographics.locationGeneral})
+  }
+  setState({
+    name: contactInfo.fullName,
+  })
+  const test = (organizations) => {
+    if (organizations) {
+      return organizations[0].title
+    } else {
```
Comment (rrgayhart) [inline app/Components/Helpers/HomeHelper.js:87]: Don't actually need an else - could be written with less lines as 

```js
if (organizations) { return organizations[0].title }
return ''
```

Comment (rrgayhart) [reply app/Components/Helpers/HomeHelper.js:87]: Or

```
return organizations ? organizations[0].title || ''
```

File: app/Components/Helpers/HomeHelper.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
+ export const checkDatabaseForSearch = (search, setState, scrubbedTweets) => {
```
Comment (rrgayhart) [inline app/Components/Helpers/HomeHelper.js:1]: The name 'helpers' becomes a 'junk drawer' - I would personally consider splitting the fetch calls and the cleaners apart and name the files a little more specifically - but that's just a suggestion

File: app/Components/Search.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  helper() {
+    checkDatabaseForSearch(this.state.search, this.setState.bind(this), this.state.scrubbedTweets)
+    this.props.history.replace('/Home')
+  }
+
+
+  render () {
+    return (
+      <article className='home-search'>
+        {/* {this.displayErrorMessage()} */}
```
Comment (rrgayhart) [inline app/Components/Search.js:24]: Do you need this?

Comment (DBULL7) [reply app/Components/Search.js:24]: It's a flash message that existed in the component I took it out of. I'm keeping it for when I bring that method over later 👍🏻

**Pull request merged**